### ISSUE TITLE
Python interface windows support

### ISSUE THREAD

**Issue by mrmbernardi** [szcompressor/SZ3#51]
Windows support is alluded to on this line: https://github.com/szcompressor/SZ3/blob/e78f4832f9327c87b0fcb53cb5f2288c2488fb65/tools/pysz/pysz.py#L21

However it doesn't work since `sys.platform` returns `win32` for Windows, not `windows`.

I have tried to use Sz3 through python on windows by patching this bug and building SZ3c with G++ and with MsBuild, but no matter what I tried I couldn't get it to work properly.

Perhaps to interface with python properly, it needs to be built with MsBuild and the functions need to be defined properly with `__declspec` like here: https://github.com/szcompressor/SZ3/blob/e78f4832f9327c87b0fcb53cb5f2288c2488fb65/tools/zstd/zstd.h#L31

**Comment by ayzk** [comment]
@mrmbernardi Thanks for pointing out the issue. I've changed the return to win32 from Windows. Have you tried the methods mentioned in this thread? https://github.com/szcompressor/SZ3/issues/5#issuecomment-1094039224

**Comment by mrmbernardi** [comment]
Yes, when build with MSYS2, UCRT64, and G++, I get library not found errors from python (this is the error when the library is found but cannot be loaded for another reason).

If I build with MsBuild, the library is successfully loaded but the names are not mangled correctly.

**Comment by ayzk** [comment]
Just to clarify, the executable can run on Win but the Python binding has problems, right?

**Comment by mrmbernardi** [comment]
When I build MsBuild, the executable works. If I build with MSYS UCRT, I get an entry point not found error.
